backend/src/main/java/com/htecgroup/skynest/repository/FileMetadataRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.htecgroup.skynest.repository;
+
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.entity.FileMetadataEntity;
+import com.htecgroup.skynest.model.entity.FolderEntity;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.UUID;
+
+public interface FileMetadataRepository extends CrudRepository<FileMetadataEntity, UUID> {
+  boolean existsByNameAndBucketAndParentFolder(
```

Comment:
This checks for a unique combination of these 3 attributes, like a composite key?

Reply:
and it works somehow

[the db will also have triggers for this purpose](https://github.com/JDP2022SkyNest/SkyNest/pull/125)